pydis_site/apps/api/serializers.py

```diff
@@ -141,30 +144,270 @@ class Meta:
         fields = ('package', 'base_url', 'inventory_url')
 
 
+#  region: filters serializers
+
+SETTINGS_FIELDS = (
+    'dm_content',
+    'dm_embed',
+    'infraction_type',
+    'infraction_reason',
+    'infraction_duration',
+    'infraction_channel',
+    'guild_pings',
+    'filter_dm',
+    'dm_pings',
+    'remove_context',
+    'send_alert',
+    'bypass_roles',
+    'enabled',
+    'enabled_channels',
+    'disabled_channels',
+    'enabled_categories',
+    'disabled_categories',
+)
+
+ALLOW_BLANK_SETTINGS = (
+    'dm_content',
+    'dm_embed',
+    'infraction_reason',
+)
+
+ALLOW_EMPTY_SETTINGS = (
+    'enabled_channels',
+    'disabled_channels',
+    'enabled_categories',
+    'disabled_categories',
+    'guild_pings',
+    'dm_pings',
+    'bypass_roles',
+)
+
+# Required fields for custom JSON representation purposes
+BASE_FILTER_FIELDS = (
+    'id', 'created_at', 'updated_at', 'content', 'description', 'additional_field'
+)
+BASE_FILTERLIST_FIELDS = ('id', 'created_at', 'updated_at', 'name', 'list_type')
+BASE_SETTINGS_FIELDS = (
+    "bypass_roles",
+    "filter_dm",
+    "enabled",
+    "remove_context",
+    "send_alert"
+)
+INFRACTION_AND_NOTIFICATION_FIELDS = (
+    "infraction_type",
+    "infraction_reason",
+    "infraction_duration",
+    "infraction_channel",
+    "dm_content",
+    "dm_embed"
+)
+CHANNEL_SCOPE_FIELDS = (
+    "disabled_channels",
+    "disabled_categories",
+    "enabled_channels",
+    "enabled_categories"
+)
+MENTIONS_FIELDS = ("guild_pings", "dm_pings")
+
+
+def _create_filter_meta_extra_kwargs() -> dict[str, dict[str, bool]]:
+    """Create the extra kwargs of the Filter serializer's Meta class."""
+    extra_kwargs = {}
+    for field in SETTINGS_FIELDS:
+        field_args = {'required': False, 'allow_null': True}
+        if field in ALLOW_BLANK_SETTINGS:
+            field_args['allow_blank'] = True
+        if field in ALLOW_EMPTY_SETTINGS:
+            field_args['allow_empty'] = True
+        extra_kwargs[field] = field_args
+    return extra_kwargs
+
+
+def get_field_value(data: dict, field_name: str) -> Any:
+    """Get the value directly from the key, or from the filter list if it's missing or is None."""
+    if data.get(field_name):
+        return data[field_name]
+    return getattr(data["filter_list"], field_name)
```

Comment:
```suggestion
def get_field_value(data: dict, field_name: str) -> Any:
    """Get the value directly from the key, or from the filter list if it's missing or is None."""
    if data.get(field_name) is None:
        return data[field_name]
    return getattr(data["filter_list"], field_name)
```

This should be checked to be `None`, to avoid a value being falsy or even `False`. Take the example of a filter with the value `enabled=False`. in this case, the `filter_list`'s `enabled` field would be used instead.

Reply:
wouldn't we want `is not None` here? So if it's not None return it, else return the filter list default

Reply:
Maybe, lol. You are correct. I've amended the suggestion. Thankyou.

Reply:
Good spot [be854fa](https://github.com/python-discord/site/pull/861/commits/be854fa3d34dac7b4b9e96b3736dd61d972f1b79)